src/SonarQube.Client/Api/SonarQubeService.cs

```diff
@@ -0,0 +1,252 @@
+
+            var credentialResponse = await InvokeRequestAsync<IValidateCredentialsRequest, bool>(token);
+            if (!credentialResponse)
+            {
+                IsConnected = false;
+                throw new InvalidOperationException("Invalid credentials");
+            }
+        }
+
+        public void Disconnect()
+        {
+            IsConnected = false;
+            httpClient.Dispose();
+        }
+
+        private void EnsureIsConnected()
+        {
+            if (!IsConnected)
+            {
+                throw new InvalidOperationException("This operation expects the service to be connected.");
+            }
+        }
+
+        public async Task<IList<SonarQubeOrganization>> GetAllOrganizationsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetOrganizationsRequest, SonarQubeOrganization[]>(
+                request =>
+                {
+                    request.OnlyUserOrganizations = true;
+                },
+                token);
+
+        public async Task<IList<SonarQubePlugin>> GetAllPluginsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPluginsRequest, SonarQubePlugin[]>(token);
+
+        public async Task<IList<SonarQubeProject>> GetAllProjectsAsync(string organizationKey, CancellationToken token) =>
+            await InvokeRequestAsync<IGetProjectsRequest, SonarQubeProject[]>(
+                request =>
+                {
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+        public async Task<IList<SonarQubeProperty>> GetAllPropertiesAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPropertiesRequest, SonarQubeProperty[]>(token);
+
+        public Uri GetProjectDashboardUrl(string projectKey)
+        {
+            EnsureIsConnected();
+
+            const string ProjectDashboardRelativeUrl = "dashboard/index/{0}";
+
+            return new Uri(httpClient.BaseAddress, string.Format(ProjectDashboardRelativeUrl, projectKey));
+        }
+
+        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey, SonarQubeLanguage language, CancellationToken token)
+        {
+            var qualityProfiles = await InvokeRequestAsync<IGetQualityProfilesRequest, SonarQubeQualityProfile[]>(
+                request =>
+                {
+                    request.ProjectKey = projectKey;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+            // Consider adding the language filter to the request configuration above and removing this line
+            var profilesWithGivenLanguage = qualityProfiles.Where(x => x.Language == language.Key).ToList();
+
+            var qualityProfile = profilesWithGivenLanguage.Count > 1
+                ? profilesWithGivenLanguage.Single(x => x.IsDefault)
+                : profilesWithGivenLanguage.Single();
```

Comment:
This is going to fail if the language isn't installed on the server (e.g. we ask for VB.NET when running against a community edition of SQ server). Is that ok? i.e. is it up to the caller to check that the appropriate plugin exists first? If so, we should document this.

Reply:
I would prefer to fix this bug in a separate PR (with a corresponding ticket) because it already exists in SLVS (this is a copy of the existing code).

Reply:
#662